static/app/components/events/interfaces/frame/stacktraceLink.tsx

```diff
@@ -100,6 +101,39 @@ function StacktraceLinkSetup({organization, project, event}: StacktraceLinkSetup
   );
 }
 
+interface CodecovLinkProps {
+  sourceUrl: string | undefined;
+  codecovStatusCode?: CodecovStatusCode;
+}
+
+function CodecovLink({sourceUrl, codecovStatusCode}: CodecovLinkProps) {
+  if (codecovStatusCode === CodecovStatusCode.NO_COVERAGE_DATA) {
+    return (
+      <CodecovWarning>
+        <span style={{padding: '0 4px'}}>{t('Code Coverage not found')}</span>
+        <StyledIconWrapper>
+          <IconWarning size="xs" color="errorText" />
+        </StyledIconWrapper>
+      </CodecovWarning>
+    );
+  }
+
+  if (codecovStatusCode === CodecovStatusCode.COVERAGE_EXISTS) {
+    const codecovUrl = sourceUrl?.replaceAll(
+      new RegExp('github.com/.[^/]*', 'g'),
+      'app.codecov.io/gh'
+    );
+
+    return (
+      <OpenInLink href={`${codecovUrl}`} openInNewTab>
```

Comment:
once sourceUrl always exists
```suggestion
      <OpenInLink href={codecovUrl} openInNewTab>
```